app/views/admin/events/_all_events.html.erb

```diff
@@ -35,7 +35,7 @@
             <% end %>
         </td> 
         <td><%= link_to 'Edit', edit_admin_event_path(event), class: 'btn btn-xs btn-warning' %></td>
-        <td><%= link_to 'Delete', admin_event_path(event), method: :delete, data: { confirm: 'Are you sure?' }, class: 'btn btn-xs btn-danger' %></td>
+        <td><%= link_to 'Delete', admin_event_path(event), method: :delete, data: ( event.published == true ? { confirm: 'Are you sure? This event has been published.' } : { confirm: 'Are you sure?'} ), class: 'btn btn-xs btn-danger' %></td>
```

Comment:
move "confirmation" block to helper, it's not correct to write equal 'true' in the conditions, just write event.published